packages/babel-helper-replace-supers/src/index.js

```diff
@@ -170,8 +181,8 @@ export default class ReplaceSupers {
     const path = opts.methodPath;
 
     this.methodPath = path;
-    this.isStatic =
-      path.isClassMethod({ static: true }) || path.isObjectMethod();
+    this.isStatic = path.isObjectMethod() || path.node.static;
+    this.isPrivateMethod = path.isPrivate() && path.isMethod();
```

Comment:
Nit: I would just add this as a new case to `isStatic`.

Reply:
I kept them separate because rolling `isPrivateMethod` into `isStatic` will cause an issue with the loose handler. If the path is an _instance_ private method, [this](https://github.com/babel/babel/blob/673385038571b0d13b6bd9efafe6206b61f0c3eb/packages/babel-helper-replace-supers/src/index.js#L151) condition becomes truthy, and [this output](https://github.com/babel/babel/blob/673385038571b0d13b6bd9efafe6206b61f0c3eb/packages/babel-plugin-proposal-private-methods/test/fixtures/private-method-loose/super/output.js#L24-L26) is instead transformed to:

```js
var _privateMethod2 = function _privateMethod2() {
  return Base.superMethod.call(this);
};
```